invenio_app_ils/records_relations/indexer.py

```diff
@@ -40,16 +42,19 @@ def index(self, record, *records):
         referenced = set()
         indexed = dict(pid_type=record._pid_type, record=record)
 
-        def add_referenced(pid, pid_type):
-            if not (pid == record["pid"] and pid_type == record._pid_type):
-                referenced.add((pid, pid_type))
+        def add_referenced(pid_value, pid_type):
+            same_record = (
+                pid_value == record["pid"] and pid_type == record._pid_type
```

Comment:
I might be wrong here, but I see we compare it only with our record, shouldn't we check if it is already in our relations list?

Reply:
added the check

Reply:
on top of that, I think we are indexing too much, I will add a comment.

Reply:
See: https://github.com/inveniosoftware/invenio-app-ils/pull/784/files#diff-b8f4363cf45e68831e5819f1540f36b8R55